controllers/jujumachine_controller.go

```diff
@@ -100,6 +124,25 @@ func (r *JujuMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{RequeueAfter: requeueTime}, nil
 	}
 
+	// Initialize the patch helper.
+	patchHelper, err := patch.NewHelper(jujuMachine, r.Client)
```

Comment:
resource is now updated at the end of each reconcile, this seems to be a common pattern instead of updating and requeuing throughout the call